lib/appcanary/configuration.rb

```diff
@@ -4,18 +4,46 @@ module Appcanary
   APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3"
 
   class Configuration
-    attr_accessor :base_uri, :api_token, :monitor_name
+    attr_accessor :base_uri, :api_key, :monitor_name, :gemfile_lock
 
     def initialize
       self.base_uri = APPCANARY_DEFAULT_BASE_URI
+      self.monitor_name = maybe_guess_monitor_name
+      self.gemfile_lock = locate_gemfile_lock
+    end
+
+    def maybe_guess_monitor_name
+      Rails.application.class.parent_name if defined?(Rails)
+    end
+
+    def locate_gemfile_lock
+      if defined?(Bundler)
+        Bundler.default_lockfile
+      else
+        begin
+          require "bundler"
+          locate_gemfile_lock
+        rescue LoadError
+          # ignore, handle at resolution time
+          return
+        end
+      end
     end
 
     def valid?
-      ! (base_uri.nil? || api_token.nil? || monitor_name.nil?)
+      ! (base_uri.nil? || api_key.nil? || gemfile_lock.nil?)
     end
   end
 
   class ConfigurationError < RuntimeError
+    def initialize(msg)
+      super(msg + <<-FOOTER)
```

Comment:
we discussed this in person but:

a) I was surprised this was valid syntax!
b) just make it a const :P 
c) it mildly breaks my syntax highlighting ;)

![screen shot 2017-01-06 at 10 30 04 am](https://cloud.githubusercontent.com/assets/17770/21722571/3a1203b2-d3fb-11e6-8104-5267676c3ed6.png)